components/breadcrumbs/breadcrumbs.js

```diff
@@ -0,0 +1,116 @@
+import './breadcrumb.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class BreadCrumbs extends RtlMixin(LitElement) {
+	static get properties() {
+		return {
+			compact: {
+				type: Boolean,
+				reflect: true
+			}
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				--d2l-breadcrumbs-background-color: rgb(251, 252, 252);
+				display: block;
+				position: relative;
+			}
+
+			:host::after {
+				background: -webkit-gradient(linear, left top, right top, from(rgba(255, 255, 255, 0)), to(var(--d2l-breadcrumbs-background-color)));
+				background: -webkit-linear-gradient(left, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				background: -o-linear-gradient(left, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				background: linear-gradient(to right, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				bottom: 0;
+				content: '';
+				pointer-events: none;
+				position: absolute;
+				right: 0;
+				top: 0;
+				width: 10px;
+			}
+
+			:host([dir="rtl"])::after {
+				background: -webkit-gradient(linear, right top, left top, from(rgba(255, 255, 255, 0)), to(var(--d2l-breadcrumbs-background-color)));
+				background: -webkit-linear-gradient(right, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				background: -o-linear-gradient(right, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				background: linear-gradient(to left, rgba(255, 255, 255, 0), var(--d2l-breadcrumbs-background-color));
+				left: 0;
+				right: auto;
+			}
+
+			.d2l-breadcrumbs-wrapper {
+				overflow: hidden;
+				position: relative;
+				white-space: nowrap;
+			}
+
+			:host([compact]) .d2l-breadcrumbs-wrapper ::slotted(d2l-breadcrumb:not(:last-of-type)) {
+				display:none;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.compact = false;
+		this._setChildrenCompactProp();
+	}
+
+	firstUpdated(changedProperties) {
+		if (changedProperties.has('compact')) {
+			this._setChildrenCompactProp();
+		}
+	}
+
+	render() {
+		return html`
+			<div class="d2l-breadcrumbs-wrapper">
+				<slot @slotchange="${this._handleSlotChange}"></slot>
+			</div>
+		`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('compact')) {
+			this._setChildrenCompactProp();
+		}
+	}
+
+	_findBreadCrumbs(nodes) {
+		let breadCrumbs = nodes.filter(item => item.nodeName === 'D2L-BREADCRUMB');
+		nodes.forEach(node => {
+			if (node.children) {
+				breadCrumbs = breadCrumbs.concat(this._findBreadCrumbs([...node.children]));
+			}
+		});
+		return breadCrumbs;
+	}
+
+	_handleSlotChange() {
+		this._setChildrenCompactProp();
+	}
+
+	_setChildrenCompactProp() {
```

Comment:
@dbatiste @AllanKerr and I chatted a bit about what to recommend here. We think moving this to the child `d2l-breadcrumb` is a better approach for now -- you're correct that it won't get updates to the parent's attribute, but we think that's worth the trade-off to avoid walking the assigned nodes in the shadow root.

There are 2 ways you could accomplish this from the child:
1. Use `findComposedAncestor` (from our DOM helpers in core) in `connectedCallback` -- this is what the old Polymer version was doing
2. Fire an event from the child in `connectedCallback` and then have the parent attach a value for `composed` to the event detail. Since events are synchronous, the child can then access that value on the next line.

Reply:
@dlockhart which one do we like more ?

Reply:
@AllanKerr @dlockhart @dbatiste So I went with the event processing way. with just minor different with what Dave suggested. Having the event being called in constructor doesn't seem to work properly as the constructors don't happen in any specific order so the event gets missed sometimes. I moved it to firstUpdated instead

Reply:
Did you try `connectedCallback`, or did that cause some trouble too?

Reply:
yeah both have the same issue

Reply:
In that case, I think Dave's original solution of using `findComposedAncestor` would be preferable, since it avoids the double-render of using `firstUpdated`. Haha sorry...

Reply:
Actually it might be the issue with console.log not being able to track the changes properly (at least the way I do it). It seems to be working fine in practice

Reply:
I moved it to connectedCallback . (Also I don't mind doing it the other way as well)